cmd/build.go

```diff
@@ -37,34 +39,30 @@ var supportedBackends = []string{
 	backendKaniko,
 }
 
-func cmdBuild(cmd *cli.Cmd) {
-	var opts buildOpts
-
-	defaultOpts(&opts, cmd)
-	cmd.BoolOptPtr(&opts.dryRun, "dry-run", false, "Simulate what would happen without actually doing anything dangerous.")
-	cmd.BoolOptPtr(&opts.forceRebuild, "force-rebuild", false, "Forces rebuilding the entire image graph, without regarding if the target version already exists.") //nolint:lll
-	cmd.BoolOptPtr(&opts.disableGenerateGraph, "no-graph", false, "Disable generation of graph during the build process.")                                          //nolint:lll
-	cmd.BoolOptPtr(&opts.disableRunTests, "no-tests", false, "Disable execution of tests during the build process.")                                                //nolint:lll
-	cmd.BoolOptPtr(&opts.disableJunitReports, "no-junit", false, "Disable generation of junit reports when running tests")
-	cmd.BoolOptPtr(&opts.retagLatest, "retag-latest", false, "Should images be retagged with the 'latest' tag for this build") //nolint:lll
-	cmd.BoolOptPtr(&opts.localOnly, "local-only", false, "Build docker images locally, do not push on remote registry")
-	cmd.StringOptPtr(&opts.backend, "b backend", backendDocker, fmt.Sprintf("Build backend used to run image builds. Supported backends: %v", supportedBackends)) //nolint:lll
-
-	cmd.Action = func() {
-		if opts.backend == backendKaniko && opts.localOnly {
+// buildCmd represents the build command
+var buildCmd = &cobra.Command{
+	Use:   "build",
+	Short: "Run docker images builds",
+	Long: `dib build will compute the graph of images, and compare it to the last built state
+
+For each image, is any file in its docker has changed, the image will be rebuilt.
+Otherwise, dib will create a new tag based on the previous tag`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		if viper.GetString(keyBackend) == backendKaniko && viper.GetBool(keyLocalOnly) {
 			logrus.Warnf("Using backend \"kaniko\" with the --local-only flag is partially supported.")
 		}
 
-		if opts.backend == backendDocker {
+		if viper.GetString(keyBackend) == backendDocker {
```

Comment:
We need to do the same logic for Kaniko

Reply:
RunPreflightChecks you mean ? I'm not sure there is any requisite for kaniko, since it only call the kube API.

Moreover, it's clearly out of scope of this MR. I'm just taking the code as is and changing the cli interface